modules/hub/scripts/gke_hub_registration.sh

```diff
@@ -33,4 +35,20 @@ trap "rm -rf $tmp_file" EXIT
 base64 --help | grep "\--decode" && B64_ARG="--decode" || B64_ARG="-d"
 echo "${SERVICE_ACCOUNT_KEY}" | base64 ${B64_ARG} > "$tmp_file"
 
-gcloud container hub memberships register "${MEMBERSHIP_NAME}" --gke-cluster="${CLUSTER_LOCATION}"/"${CLUSTER_NAME}" --service-account-key-file="${tmp_file}" --project="${PROJECT_ID}" --quiet
+if [[ ${GKE_CLUSTER_FLAG} == 1 ]]; then
+    echo "Registering GKE Cluster."
+    gcloud container hub memberships register "${MEMBERSHIP_NAME}" --gke-cluster="${CLUSTER_LOCATION}"/"${CLUSTER_NAME}" --service-account-key-file="${tmp_file}" --project="${PROJECT_ID}" --quiet
+else
+    echo "Registering a non-GKE Cluster. Using current-context to register Hub membership."
+    #Get the kubeconfig
+    CONTEXT=$(kubectl config current-context)
+    gcloud container hub memberships register "${MEMBERSHIP_NAME}" --context="${CONTEXT}" --service-account-key-file="${tmp_file}" --project="${PROJECT_ID}" --quiet
+fi
+
+
+# Add labels to the registered cluster
+if [ -z ${LABELS+x} ]; then
+    echo "No hub labels to apply."
+else
+    gcloud container hub memberships update "${MEMBERSHIP_NAME}" --update-labels "$LABELS"
```

Comment:
It may otherwise pull from gcloud config.
```suggestion
    gcloud container hub memberships update "${MEMBERSHIP_NAME}" --update-labels "$LABELS" --project="${PROJECT_ID}"
```